runners/direct-java/src/test/java/org/apache/beam/runners/direct/BoundedReadEvaluatorFactoryTest.java

```diff
@@ -115,6 +123,37 @@ public void boundedSourceInMemoryTransformEvaluatorProducesElements() throws Exc
   }
 
   @Test
+  public void getInitialInputsSplitsIntoBundles() throws Exception {
+    when(context.createRootBundle())
+        .thenAnswer(
+            new Answer<UncommittedBundle<?>>() {
+              @Override
+              public UncommittedBundle<?> answer(InvocationOnMock invocation) throws Throwable {
+                return bundleFactory.createRootBundle();
+              }
+            });
+    Collection<CommittedBundle<?>> initialInputs =
+        new BoundedReadEvaluatorFactory.InputProvider(context)
+            .getInitialInputs(longs.getProducingTransformInternal(), 3);
+
+    assertThat(initialInputs, hasSize(allOf(greaterThanOrEqualTo(2), lessThanOrEqualTo(4))));
```

Comment:
Because the source measures in terms of bytes, the `CountingSource` may produce additional bundles (because you can't evenly fit 10 longs into three "equally-sized" bundles.

I have tightened the bound, however.